### PR TITLE
Tsne dx native

### PR DESCRIPTION
**Pull request by adthrasher**
Updating RNA-Seq Expression Classification documentation to reflect the DX native app

### CODE REVIEW

**Comment by adthrasher** [comment]
@agout - Can you review as well?

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 1]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
+
+```bash
+# you can run the following code snippet after filling in your values
+# to successfully prepare the file.
+
+file_id=<DNAnexus file ID> # file ID or file path
+dx set_properties $file_id sample_name="<value>" # Should match the file name up to the first period character
+dx set_properties $file_id strandedness="<value>" # Stranded-Forward, Stranded-Reverse, or Unstranded
+dx set_properties $file_id library_type="<value>" # PolyA or Total
+dx set_properties $file_id read_length="<value>" # Integer number of base pairs in reads
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:121]: Potentially we should put `(e.g. "101")` in the comment.

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 2]
```diff
@@ -114,39 +169,48 @@ Once the resulting analysis job completes, an HTML plot of the results should be
...
 ## Known issues
 
-!!! caution "Fresh Frozen Tissue Only
-     The RNA-Seq Expression Classification pipeline reference data uses sequencing data from fresh, frozen 
-     tissue samples. It has not been evaluated for use with sequencing data 
-     generated from formalin-fixed paraffin-embedded (FFPE) specimens. 
+There are a few known cautions with the RNA-Seq Expression Classification workflow.
+
+!!! caution "Data must fit well defined values"
+    The RNA-Seq Expression Classification pipeline reference data is based on GRCh38 aligned, Gencode v31 annotated samples from fresh, frozen tissue samples. It has not been evaluated for samples not meet this criteria.
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:194]:
```suggestion
    The RNA-Seq Expression Classification pipeline reference data is based on GRCh38 aligned, Gencode v31 annotated samples from fresh, frozen tissue samples. It has not been evaluated for samples that do not meet this criteria.
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 3]
```diff
@@ -2,15 +2,15 @@
 
 ## Overview
 
-St. Jude Cloud provides functionality for generating RNA-Seq Expression Classification plots. This tool allows plotting of RNA-seq data by running through the St. Jude Cloud normalization [pipeline](https://stjudecloud.github.io/rfcs/0001-rnaseq-workflow-v2.0.html). The generated count data is then compared to a reference set of data from a cohort of St. Jude samples and a plot is produced. 
+St. Jude Cloud provides functionality for generating RNA-Seq Expression Classification plots. This tool allows plotting of RNA-seq data by running through the St. Jude Cloud normalization [pipeline](https://stjudecloud.github.io/rfcs/0001-rnaseq-workflow-v2.0.html). The generated count data is then compared to a reference set of data from a cohort of St. Jude samples and a plot is produced.
 
 ## Requirements
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:7]:
```suggestion
### Requirements
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 4]
```diff
@@ -2,15 +2,15 @@
...
+* The RNA-Seq Expression Classification pipeline reference data uses sequencing data from fresh, frozen tissue samples. It has not been evaluated for use with sequencing data generated from formalin-fixed paraffin-embedded (FFPE) specimens.
+* If running the count-based pipeline, alignment must be done against the [GRCh38_no_alt reference](ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz). It should use parameters as specifed in our [RNA-seq workflow](https://stjudecloud.github.io/rfcs/0001-rnaseq-workflow-v2.0.0.html) to minimize any discrepancies caused by differing alignment specification.
+* If running the count-based pipeline, feature counts should be generated with htseq-count as described in our [RNA-seq workflow](https://stjudecloud.github.io/rfcs/0001-rnaseq-workflow-v2.0.0.html). This pipeline uses [Gencode v31](ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/gencode.v31.annotation.gtf.gz) annotations.
 
-## Inputs 
+## Inputs
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:13]:
```suggestion
### Inputs
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 5]
```diff
@@ -21,12 +21,12 @@ with a counts file or a BAM file.
...
     it **must** be aligned to `GRCh38_no_alt`.
-    Running a BAM aligned to any other reference genome is not supported. Maybe more
+    Running a sample aligned to any other reference genome is not supported. Maybe more
     importantly, we do not check the genome build of the BAM, so errors in computation
-    or the results can occur. If your BAM is *not* aligned to this genome build, we 
-    recommend submitting the BAM to the realignment-based workflow.
+    or the results can occur. If your sample is *not* aligned to this genome build, we
+    recommend submitting the sample to the realignment-based workflow.
 
 ## Outputs
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:31]:
```suggestion
### Outputs
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 6]
```diff
@@ -36,27 +36,25 @@ The RNA-Seq Expression Classification pipeline produces the following outputs:
 |--|--|--|
 | Interactive expression plot (.html) | Visualization of RNA-Seq data | all |
 | Aligned BAM (.bam) | BAM file produced by our RNA-Seq pipeline for the input samples. | Realignment |
-| Feature read counts (.txt) | Read counts for the Gencode features. | BAM-based and Realignment |
+| Feature read counts (.txt) | Read counts for the Gencode features. | Realignment |
 
-## Workflow Steps 
+## Workflow Steps
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:41]:
```suggestion
### Workflow Steps
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 7]
```diff
@@ -36,27 +36,25 @@ The RNA-Seq Expression Classification pipeline produces the following outputs:
...
    using [standard STAR mapping](https://stjudecloud.github.io/rfcs/0001-rnaseq-workflow-v2.0.0.html).
-   * Only for realignment workflow
-2. A feature count (.count.txt) file is produced for comparison to 
+2. [Only for realignment workflow] A feature count (.count.txt) file is produced for comparison to
    St. Jude Cloud reference data.
-   * For BAM-based and realignment workflows
-3. A visualization for genomic features is produced. 
+3. A visualization for genomic features is produced.
 
 ### Mapping
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:49]:
```suggestion
#### Mapping
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 8]
```diff
@@ -36,27 +36,25 @@ The RNA-Seq Expression Classification pipeline produces the following outputs:
...
 
-We use the [STAR aligner](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/) 
-to rapidly map reads to the GRCh38 human reference genome. 
+We use the [STAR aligner](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/)
+to rapidly map reads to the GRCh38 human reference genome.
 
 ### Feature Counts
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:54]:
```suggestion
#### Feature Counts
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 9]
```diff
@@ -36,27 +36,25 @@ The RNA-Seq Expression Classification pipeline produces the following outputs:
...
 
-We use [htseq-count](https://htseq.readthedocs.io/en/latest/count.html) 
-to produce genomic feature counts. 
+We use [htseq-count](https://htseq.readthedocs.io/en/latest/count.html)
+to produce genomic feature counts.
 
 ### Visualization
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:59]:
```suggestion
#### Visualization
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 10]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
 ## Getting Started
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:63]:
```suggestion
## Running the Workflow

### Getting Started
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 11]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
+    or the results can occur. If your sample is *not* aligned to this genome build, we
+    recommend submitting the sample to the realignment-based workflow.
 
-We provide three versions of the RNA-Seq Expression Classification tool depending on desired input. The full workflow allows a user to upload a sample in BAM format. That sample will then be converted to `FastQ` format, aligned with `STAR` two-pass alignment, and feature counts generated with `htseq-count`
+We provide two versions of the RNA-Seq Expression Classification tool depending on desired input. The full workflow allows a user to upload a sample in BAM format. That sample will then be converted to `FastQ` format, aligned with `STAR` two-pass alignment, and feature counts generated with `htseq-count`
 
 To get started, you need to navigate to the [RNA-Seq Expression Classification tool page](https://platform.stjude.cloud/tools/rnaseq-expression-classification). You'll need to click the "Start" button in the left hand pane. This creates a cloud workspace in DNAnexus with the same name as the tool. After this, you will be able to upload your input files to that workspace.
 
 ## Uploading data
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:77]:
```suggestion
### Uploading data
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 12]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
-The RNA-Seq Expression Classification pipeline takes either a htseq-count count file or a GRCh38_no_alt aligned BAM from human RNA-Seq. You can upload your input file(s) using the data transfer application or by uploading them through the command line. Both of the guides linked here will contain more details on how to upload data using that method, so we defer to those guides here.
+The RNA-Seq Expression Classification pipeline takes either a htseq-count count file or a GRCh38_no_alt aligned BAM from human RNA-Seq. You can upload your input file(s) through the command line. See [Uploading Data from your Local Computer](../../../covid-19/upload-local).
+
+Once you have the `dx` doolkit, to upload a sample HTSeq count file `sample.counts.txt` to the `inputs` folder of the `project-rnaseq` cloud project, you could use the following command:
 
+```bash
+dx upload sample.counts.txt --destination "project-rnaseq:/inputs/"
+```
 
 ## Running the tool
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:87]:
```suggestion
### Running the tool
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 13]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
 
 Once you've uploaded data to your cloud workspace, click "Launch Tool" on the tool's landing page. A dropdown will present the different presets for running the workflow. Here, you can select whether you wish to start with a counts file or a BAM file.
 
+## Obtaining reference data
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:91]:
```suggestion
### Obtaining reference data
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 14]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
+
+Reference feature count data can be retrieved through the [Genomics Platform Data Browser](https://platform.stjude.cloud/data/publications?publication_accession=SJC-PB-1020). 
+
+![](../../../images/guides/genomics-platform/analyzing-data/sj-workflows/rnaseq-expression-classification/Requesting-StJudeCloud-paper.gif)
+
+These must then be provided to the workflow through the `reference_counts` parameter. By default, all reference files will be used by the app, but this can be restricted to one of the three tumor types [Blood, Brain, Solid] through the app settings.
+
+## Preparing input data
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:99]:
```suggestion
### Preparing input data
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 15]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
+dx set_properties $file_id library_type="<value>" # PolyA or Total
+dx set_properties $file_id read_length="<value>" # Integer number of base pairs in reads
+dx set_properties $file_id pairing="<value>" # Paired-end or Single-end
+```
+
+The file ID can be retrieved from the DNAnexus web interface. Click on the file of interest and the file ID is displayed in the sidebar.
+
+![](../../../images/guides/genomics-platform/analyzing-data/sj-workflows/rnaseq-expression-classification/file_id.png)
+
 ## Hooking up inputs
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:129]:
```suggestion
### Hooking up inputs
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 16]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
 
-![](../../../images/guides/genomics-platform/analyzing-data/sj-workflows/rnaseq-expression-classification/tsne-param.png)
+![](../../../images/guides/genomics-platform/analyzing-data/sj-workflows/rnaseq-expression-classification/rnaseq-expression-workflow-options-new-ui.png)
+
+If running the realignment workflow, the input file should be specified as a BAM to realign. HTSeq will run on the realigned BAM and the result will be passed into the t-SNE app for plotting against the reference data. The reference data should be specified in the `reference_counts` parameter as an array of HTSeq count files. The BAM will be specified to the RNA-Seq V2 stage as `input_bam`. The BAM should have properties set as described above. These will automatically be applied to the new HTSeq count file.
+
+![](../../../images/guides/genomics-platform/analyzing-data/sj-workflows/rnaseq-expression-classification/rnaseq-expression-workflow-realignment-options-new-ui.png)
 
 ## Starting the workflow
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:153]:
```suggestion
### Starting the workflow
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 17]
```diff
@@ -65,39 +63,96 @@ A t-Distributed Stochastic Neighbor Embedding (t-SNE) visualization is produced
...
 
-Once your input files are hooked up, you should be able to start the workflow by clicking the "Run as Analysis..." button in the top right hand corner of the workflow dialog.
+Once your input files are hooked up, you should be able to start the workflow by clicking the "Run Analysis" button in the top right hand corner of the workflow dialog.
 
 ## Monitoring run progress
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:157]:
```suggestion
### Monitoring run progress
```

File: docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md [hunk 18]
```diff
@@ -114,39 +169,48 @@ Once the resulting analysis job completes, an HTML plot of the results should be
```
Comment (claymcleod) [inline docs/guides/genomics-platform/analyzing-data/rnaseq-expression-classification.md:169]: Can't see it on the diff, but `## Cost Estimation` should become `### Cost Estimation` and we should consider moving it to the last section in Overview.

**Review by claymcleod — APPROVED** [review]
Looks good! Please also add all of these typo checks to be ignored in https://github.com/stjudecloud/docs/blob/master/.typo-ci.yml

**Review by claymcleod — APPROVED** [review]

**Pull request merged**